src/Utility/RerunTotalStatistics.php

```diff
@@ -0,0 +1,308 @@
+    /**
+     * @var HookStat[]
+     */
+    private $failedHookStats = array();
+    /**
+     * @var StepStat[]
+     */
+    private $passedStepStats = array();
+
+    /**
+     * Initializes statistics.
+     */
+    public function __construct()
+    {
+        $this->resetAllCounters();
+
+        $this->timer = new Timer();
+        $this->memory = new Memory();
+    }
+
+    public function resetAllCounters()
+    {
+        $this->scenarioCounters = $this->stepCounters = array(
+            TestResult::PASSED    => 0,
+            TestResult::FAILED    => 0,
+            StepResult::UNDEFINED => 0,
+            TestResult::PENDING   => 0,
+            TestResult::SKIPPED   => 0
+        );
+    }
+
+    /**
+     * Starts timer.
+     */
+    public function startTimer()
+    {
+        $this->timer->start();
+    }
+
+    /**
+     * Stops timer.
+     */
+    public function stopTimer()
+    {
+        $this->timer->stop();
+    }
+
+    /**
+     * Returns timer object.
+     *
+     * @return Timer
+     */
+    public function getTimer()
+    {
+        return $this->timer;
+    }
+
+    /**
+     * Returns memory usage object.
+     *
+     * @return Memory
+     */
+    public function getMemory()
+    {
+        return $this->memory;
+    }
+
+    /**
+     * Registers scenario stat.
+     *
+     * @param ScenarioStat $stat
+     */
+    public function registerScenarioStat(ScenarioStat $stat)
+    {
+        if (TestResults::NO_TESTS === $stat->getResultCode()) {
+            return;
+        }
+
+        $this->scenarioCounters[$stat->getResultCode()]++;
+
+        if (TestResult::FAILED === $stat->getResultCode()) {
+            // Ensure that any scenario reruns aren't counted as additional failures
+            $alreadyHasFailure = false;
+            foreach ($this->failedScenarioStats as $failedStat) {
+                if ($failedStat->getPath() === $stat->getPath()) {
+                    $alreadyHasFailure = true;
+                    break;
+                }
+            }
+            if (!$alreadyHasFailure) {
+                $this->failedScenarioStats[] = $stat;
+            } else {
+                $this->scenarioCounters[TestResult::FAILED]--;
+            }
+        }
+
+        if (TestResult::PASSED == $stat->getResultCode()) {
+            // Remove the scenario from the failed scenarios list if it passes on rerun
+            $newFailedScenarioStats = [];
+            foreach ($this->failedScenarioStats as $failedStat) {
+                if ($failedStat->getPath() !== $stat->getPath()) {
+                    $newFailedScenarioStats[] = $failedStat;
+                } else {
+                    $this->scenarioCounters[TestResult::FAILED]--;
+                }
+            }
+            $this->failedScenarioStats = $newFailedScenarioStats;
+        }
+
+        if (TestResult::SKIPPED === $stat->getResultCode()) {
+            $this->skippedScenarioStats[] = $stat;
```

Comment:
If it's feasible, remove skipped items as well. Seeing this is a bit concerning:
![image](https://github.com/silverstripe/silverstripe-behat-extension/assets/36352093/28efd1ff-3811-43cb-ba4d-5a760a544046)

Also if feasible, since we're controlling this class now, can we add a "1 rerun"? That's very much not necessary so if it's at all difficult don't bother, but could be good if we want an idea of how often these are being rerun, we can just check the stats at the end of the behat log.

Reply:
I don't think either's easily feasible.  I don't see a `TestResult::SKIPPED` bit in the step method (though there is one for the scenario step), so I'm not sure where that's coming from.

The `1 rerun` stat would require dispatching a new event class, it's doable but should be done as a new card if we want it IMO